src/snyk/common/download/downloader.ts

```diff
@@ -1,53 +1,51 @@
 import axios, { CancelTokenSource } from 'axios';
 import * as fs from 'fs';
-import { mkdirSync } from 'fs';
 import * as fsPromises from 'fs/promises';
-import path from 'path';
 import * as stream from 'stream';
 import { Progress } from 'vscode';
 import { Checksum } from '../../cli/checksum';
-import { CliExecutable } from '../../cli/cliExecutable';
 import { messages } from '../../cli/messages/messages';
 import { IConfiguration } from '../configuration/configuration';
-import { LsExecutable } from '../languageServer/lsExecutable';
-import { IStaticLsApi } from '../languageServer/staticLsApi';
-import { LsSupportedPlatform } from '../languageServer/supportedPlatforms';
+import { IStaticCliApi } from '../../cli/staticCliApi';
+import { CliExecutable } from '../../cli/cliExecutable';
 import { ILog } from '../logger/interfaces';
 import { CancellationToken } from '../vscode/types';
 import { IVSCodeWindow } from '../vscode/window';
+import { CliSupportedPlatform } from '../../cli/supportedPlatforms';
+import { ExtensionContext } from '../vscode/extensionContext';
 
 export type DownloadAxiosResponse = { data: stream.Readable; headers: { [header: string]: unknown } };
 
 export class Downloader {
   constructor(
     private readonly configuration: IConfiguration,
-    private readonly lsApi: IStaticLsApi,
+    private readonly cliApi: IStaticCliApi,
     private readonly window: IVSCodeWindow,
     private readonly logger: ILog,
+    private readonly extensionContext: ExtensionContext,
   ) {}
-
   /**
    * Downloads LS. Existing executable is deleted.
    */
-  async download(): Promise<CliExecutable | LsExecutable | null> {
-    const lsPlatform = LsExecutable.getCurrentWithArch();
-    if (lsPlatform === null) {
+  async download(): Promise<CliExecutable | null> {
+    const platform = await CliExecutable.getCurrentWithArch();
+    if (platform === null) {
       return Promise.reject(!messages.notSupported);
     }
-    return await this.getLsExecutable(lsPlatform);
+    return await this.getCliExecutable(platform);
   }
 
-  private async getLsExecutable(lsPlatform: LsSupportedPlatform): Promise<LsExecutable | null> {
-    const lsPath = LsExecutable.getPath(this.configuration.getSnykLanguageServerPath());
-    const lsDir = path.dirname(lsPath);
-    mkdirSync(lsDir, { recursive: true });
-    if (await this.binaryExists(lsPath)) {
-      await this.deleteFileAtPath(lsPath);
+  private async getCliExecutable(platform: CliSupportedPlatform): Promise<CliExecutable | null> {
+    const cliPath = await CliExecutable.getPath(
+      this.extensionContext.extensionPath,
+      await this.configuration.getCliPath(),
+    );
+    if (await this.binaryExists(cliPath)) {
```

Comment:
here, the cli path is used including the binary - above it is set to default to the extension path. This feels like a potential bug.

Reply:
https://github.com/snyk/vscode-extension/pull/551#discussion_r1822112642